components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -436,46 +454,44 @@ - (void)setBackgroundColor:(UIColor *)backgroundColor {
-    [self fhv_adjustInsetsForSafeAreaInScrollView:_trackingScrollView];
-
-    // If the min/max height hasn't been explicitly set, we need to update the layout to reflect
-    // the change in the Safe Area insets.
-    if (_hasExplicitlySetMinHeight && _hasExplicitlySetMaxHeight) {
-      return;
-    }
-
     if (_isChangingStatusBarVisibility) {
       // We aren't interest in safe area inset changes due to status bar visibility changes - we're
       // only interested in hardware-related safe area changes. If we know that we're changing the
       // status bar visibility then we ignore this safeAreaInsetsDidChange event.
       return;
     }
-
-    if (!_hasExplicitlySetMinHeight) {
-      // Edge case for UITableViewController: If we're a subview of _trackingScrollView,
-      // we need to get the Safe Area insets from there and not use ours.
-      if ([self isDescendantOfView:_trackingScrollView]) {
-        _minimumHeight = kFlexibleHeaderDefaultHeight + _trackingScrollView.safeAreaInsets.top;
-      } else {
-        _minimumHeight = kFlexibleHeaderDefaultHeight + self.safeAreaInsets.top;
-      }
-    }
-    if (!_hasExplicitlySetMaxHeight) {
-      _maximumHeight = _minimumHeight;
+    CGFloat safeAreaTop = [self flexibleHeaderSafeAreaTop];
+    if (_currentSafeAreaInsetTop == safeAreaTop) {
+      return;
     }
-    if (_maximumHeight < _minimumHeight) {
-      _maximumHeight = _minimumHeight;
+    _currentSafeAreaInsetTop = safeAreaTop;
+
+    // If the min or max height have been explicitly set, don't adjust anything if the values
+    // already include a Safe Area inset.
+    if (_hasExplicitlySetMinOrMaxHeight && _minMaxHeightIncludesSafeArea) {
+      return;
+    } else if (!_hasExplicitlySetMinOrMaxHeight && _minMaxHeightIncludesSafeArea) {
+      // If we're using the defaults we need to update them to account for the new Safe Area inset.
+      _minimumHeight = kFlexibleHeaderDefaultHeight + [self flexibleHeaderSafeAreaTop];
+      _maximumHeight = kFlexibleHeaderDefaultHeight + [self flexibleHeaderSafeAreaTop];;
     }
 
+    // Adjust the scroll view insets to account for the new Safe Area inset.
+    [self fhv_adjustTrackingScrollViewInsets];
+
     // Ignore any content offset delta that occured as a result of any safe area insets change.
     _shiftAccumulatorLastContentOffset = [self fhv_boundedContentOffset];
 
     // The changes might require us to re-calculate the frame, or update the entire layout.
     if (!_trackingScrollView) {
       CGRect bounds = self.bounds;
-      bounds.size.height = _minimumHeight;
+      bounds.size.height = self.computedMinimumHeight;
       self.bounds = bounds;
-      [self fhv_commitAccumulatorToFrame];
+      CGPoint position = self.center;
+      position.y = -MIN([self fhv_accumulatorMax], _shiftAccumulator);
+      position.y += self.bounds.size.height / 2;
+      self.center = position;
+      [self.delegate flexibleHeaderViewFrameDidChange:self];
```

Comment:
Can you document why we're not using fhv_commitAccumulatorToFrame here? The main difference I see is that the `_statusBarShifter` isn't being updated anymore.

Reply:
fhv_commitAccumulatorToFrame calls fhv_accumulatorDidChange, and that method sets the shadow opacity to the default opacity (0.4) even when we're not tracking a scroll view -_-